### ISSUE TITLE
Classification metrics for tf.Tensor output_keys with non-float dtype

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#665]
Sometimes we have a discrete random variable that has type `tf.Tensor` and not `ed.RandomVariable`. The tensor has discrete values and is random due to inputs somewhere in its graph being random. For example, Bayesian GANs for classification:
```python
import tensorflow as tf
from edward.models import Normal

# weights and biases have Normal(0, 1) prior
W1 = Normal(tf.zeros([501, 100]), tf.ones([501, 100]))
W2 = Normal(tf.zeros([100,   1]), tf.ones([100,   1]))
b1 = Normal(tf.zeros(100), tf.ones(100))
b2 = Normal(tf.zeros(1),   tf.ones(1))

# set up inputs to neural network
X = tf.placeholder(tf.float32, [N, 500])
eps = Normal(tf.zeros([N, 1]), tf.ones([N, 1]))

# y = neural_network([x, eps])
input = tf.concat([X, eps], 1)
h1 = tf.nn.relu(tf.matmul(input, W1) + b1)
h2 = tf.matmul(h1, W2) + b2
y = tf.reshape(tf.sign(h2), [-1])  # take sign, then flatten
```

After training, we'd like to use `ed.evaluate` with `y` as a predicted class label:
```python
ed.evaluate('binary_accuracy', data={y: y_data, X: X_data})
```
However, when we calculate `y_pred` using `tf.Tensor` output keys, we take the mean over Monte Carlo samples. This doesn't work for discrete values; we should take the mode over Monte Carlo samples instead.

**Comment by ErmiaAzarkhalili** [comment]
Hi @dustinvtran ,
Do you have any plan to add these feature to Edward in soon future?

**Comment by dustinvtran** [comment]
Haven't found it to be a pressing issue. Contributions are welcome of course.

**Comment by ErmiaAzarkhalili** [comment]
hi @dustinvtran ,
Could you please let me know how I can contribute to this part of edward? 
we discussed about it on the forum where you provided modification into evaluate module, but I did not get what you said.

Thanks.
